src/test/java/org/opensearch/security/dlic/dlsfls/CustomFieldMaskedComplexMappingTest.java

```diff
@@ -167,10 +167,10 @@ public void testComplexMappingSearch() throws Exception {
                     "88783587fef740690c4fa39476fb86314d034fa3370e1a1fa186f6d9d4644a18ad85063c1e3161f8929f7ca019bb8740611eaf337709113901e7c3a6b59f4166"
                 )
         );
-        Assert.assertFalse(res.getBody().contains("e90a2fdf7b1939ec06e294321fd7d23e1a70d8fc080a3f85d0f3bf08c205b53"));
+        Assert.assertFalse(res.getBody().contains("9fe023e13d5179157b2023c21c60bc98340a12470538affd306281619d2477c3"));
```

Comment:
> The only difference is that the Blake2b hash values will be different [...]

This seems specifically backward **in**compatible.  I would not expect any changes to these unit tests - I don't think this is safe to backport.  Can we add an alternative provider for the hashing logic that can be 'configured on' rather than on by default?  This allows system admins to choose if/when they want to take this change

Reply:
@peternied The hashing changes because the salt does not appear to be passed correctly in the existing code. Please see comment https://github.com/opensearch-project/security/issues/4212#issuecomment-2063453718